### PR TITLE
Close #21573: Constraint image size when using an icon in jump back in or recently bookmarked sections

### LINKED ISSUE #21573: Update image size for items under Recently Bookmarked and Jump back in to match Top Sites
This task is to update the image (thumbnail/favicon size) for the Recently Bookmarked and Jump Back In section to match Top Sites.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
Result

**Comment by rocketsroger** [comment]
Without my change it looks like this currently when thumbnails are unavailabl
device-2021-09-29-164439
e.

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt [hunk 1]
```diff
@@ -224,48 +229,117 @@ private fun RecentSearchGroupItem(
...
+        }
+        else -> {
+            components.core.icons.Loader(
+                url = url
+            ) {
+                Placeholder {
+                    Box(
+                        modifier = Modifier.background(
+                            color = when (isSystemInDarkTheme()) {
+                                true -> Color(0xFF42414D) // DarkGrey30
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:327]: How come an explicit color hex here?

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt [hunk 2]
```diff
@@ -224,48 +229,117 @@ private fun RecentSearchGroupItem(
...
+/**
+ * A recent tab icon.
+ *
+ * @param url The loaded URL of the tab.
+ * @param modifier modifier Modifier used to draw the image content.
+ * @param icon The icon of the tab.
+ * is null.
+ */
+@Composable
+private fun RecentTabIcon(
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:306]: Has there been any discussion around doing @Previews for our composeable layouts going forward?

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:327]: Good question.  I don't know why.  I got this from top sites.  I'll look into this.  Thanks,

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt (hunk 2 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:306]: There is. I think due to the amount of work recently that is on hold for now.  I agree that we should definitely add @Previews as soon as we can.

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt (hunk 1 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:327]: At that point, https://github.com/mozilla-mobile/android-components/pull/10954 had not landed yet, so we kept a reference to what the colour name should be when we were able to use it.

**Comment by rocketsroger** [comment]
New implementation when thumbnails are not available.
New

**Comment by rocketsroger** [comment]
Since we have the new design, I'll land this before the Beta cut.

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt [hunk 3]
```diff
@@ -326,7 +393,7 @@ private fun ThumbnailImage(
     alignment: Alignment
 ) {
     val rememberBitmap = remember(tabId) { mutableStateOf<ImageBitmap?>(null) }
-    val size = LocalDensity.current.run { 116.dp.toPx().toInt() }
+    val size = LocalDensity.current.run { 112.dp.toPx().toInt() }
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:396]:
```suggestion
    val size = LocalDensity.current.run { 108.dp.toPx().toInt() }
```

**Review by jonalmeida — APPROVED** [review]

**Comment by rocketsroger** [comment]
Confirmed with @Kate-Galetski that this is good to go.

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt (hunk 3 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt]: Good catch.  I'll fix.  Thanks,

**Pull request merged**